libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -44,3 +44,54 @@ int32_t AP_BattMonitor_Backend::get_capacity() const
 {
     return _mon.pack_capacity_mah(_state.instance);
 }
+
+// update battery resistance estimate
+// faster rates of change of the current and voltage readings cause faster updates to the resistance estimate
+// the battery resistance is calculated by comparing the latest current and voltage readings to a low-pass filtered current and voltage
+// high current steps are integrated into the resistance estimate by varying the time constant of the resistance filter
+void AP_BattMonitor_Backend::update_resistance_estimate()
+{
+    // return immediately if no current
+    if (!has_current() || _state.current_amps <= 0.0f) {
```

Comment:
I'd recommend ```!is_positive(_state.current_amps)``` instead of <= 0.0f. This would allow you to remove the ```is_zero(around _current_max_amps)``` check 7 lines from now.